opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,314 @@
+			"missing_package_result": schema.StringAttribute{
+				Description: "The check result if the package isn't being used by a service. (Optional.)",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllCheckResultStatusEnum...),
+				},
+			},
+			"package_constraint": schema.StringAttribute{
+				Description: "The package constraint the service is to be checked for. (Required.)",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllPackageConstraintEnum...),
+				},
+			},
+			"package_manager": schema.StringAttribute{
+				Description: "The package manager (ecosystem) this package relates to. (Required.)",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(opslevel.AllPackageManagerEnum...),
+				},
+			},
+			"package_name": schema.StringAttribute{
+				Description: "The name of the package to be checked. (Required.)",
+				Required:    true,
+			},
+			"package_name_is_regex": schema.BoolAttribute{
+				Description: "Whether or not the value in the package name field is a regular expression. (Optional.)",
+				Optional:    true,
+			},
+			"version_constraint_predicate": PredicateSchema(),
+		}),
+	}
+}
+
+func (r *CheckPackageVersionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel CheckPackageVersionResourceModel
+
+	// Read Terraform plan data into the planModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	input := opslevel.CheckPackageVersionCreateInput{
+		CategoryId: asID(planModel.Category),
+		Enabled:    planModel.Enabled.ValueBoolPointer(),
+		FilterId:   opslevel.RefOf(asID(planModel.Filter)),
+		LevelId:    asID(planModel.Level),
+		Name:       planModel.Name.ValueString(),
+		Notes:      planModel.Notes.ValueStringPointer(),
+		OwnerId:    opslevel.RefOf(asID(planModel.Owner)),
+
+		PackageConstraint: opslevel.PackageConstraintEnum(planModel.PackageConstraint.ValueString()),
+		PackageManager:    opslevel.PackageManagerEnum(planModel.PackageManager.ValueString()),
+		PackageName:       planModel.PackageName.ValueString(),
+	}
+	if !planModel.EnableOn.IsNull() {
+		enabledOn, err := iso8601.ParseString(planModel.EnableOn.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError("error", err.Error())
+		}
+		input.EnableOn = &iso8601.Time{Time: enabledOn}
+	}
+
+	if !planModel.MissingPackageResult.IsNull() {
+		input.MissingPackageResult = opslevel.RefOf(opslevel.CheckResultStatusEnum(planModel.MissingPackageResult.ValueString()))
+	}
+	if !planModel.PackageNameIsRegex.IsNull() {
+		input.PackageNameIsRegex = planModel.PackageNameIsRegex.ValueBoolPointer()
+	}
+	predicateModel, diags := PredicateObjectToModel(ctx, planModel.VersionConstraintPredicate)
+	resp.Diagnostics.Append(diags...)
+	if !predicateModel.Type.IsUnknown() && !predicateModel.Type.IsNull() {
```

Comment:
The validation logic here can be removed after `ValidateConfig()` is added

Reply:
Since this is how most of the other checks are written let's keep things consistent can keep what's here. Ignore these two comments 😁